picosha2.h

```diff
@@ -369,7 +373,43 @@ template <typename InContainer>
 std::string hash256_hex_string(const InContainer& src) {
     return hash256_hex_string(src.begin(), src.end());
 }
-
-}  // namespace picosha2
-
+std::string hash256_from_ifstream(std::ifstream f){
+    std::vector<char> data;
+    data.insert(data.begin(),std::istreambuf_iterator<char>(f), std::istreambuf_iterator<char>());
+    std::string hashed;
+    hash256_hex_string(std::string(data.begin(),data.end()), hashed);
+    return hashed;
+}
+inline void multithread(std::vector<std::string> src, std::deque<std::vector<std::string>>* solved){
+    std::vector<std::string> solve;
+    for(std::string i: src){
+        solve.push_back(hash256_hex_string(i));
+    }
+    solved->push_back(solve);
+}
+std::vector<std::string> hash_Vector(std::vector<std::string> src){
```

Comment:
I'm disagree to add this function. This feaature seems outside of the domain which PicoSHA2 covers.

Reply:
Parallelism for one hash (not multiple) is welcome because it is inside of the domain.

Reply:
But parallelism for one hash is imposible.

Reply:
SIMD enables it. (Now I don't have an enough motivation to make sha2 functions with SIMD but if you or others send PR with it, I will merge it with pleasure)

Reply:
When we use SIMD we are calling low level processor functions so we lose portability and we must compile it.